### PR TITLE
Fixed #32733 -- Skipped system check for specifying type of auto-created primary keys on abstract models.

### PR DESCRIPTION
**Pull request by amureki**
ticket-32733

System check was introduced in b5e12d4.

### CODE REVIEW

File: tests/check_framework/test_model_checks.py [hunk 1]
```diff
@@ -403,6 +403,18 @@ class Child(Parent):
...
 
+    def test_explicit_pk_with_abstract_parent(self):
+        class Abstract(models.Model):
+            class Meta:
+                abstract = True
+
+        class Child(Abstract):
+            id = models.AutoField(primary_key=True)
+
+        self.assertEqual(checks.run_checks(app_configs=self.apps.get_app_configs()), [])
```
Comment (codingjoe) [inline tests/check_framework/test_model_checks.py:414]: This bit wasn't broken before, right? So, there should be already tests covering this behavior.

Comment (amureki) [reply tests/check_framework/test_model_checks.py:414]: Well, it was intended to test that child of an abstract parent also works fine. :)
But if it does not make sense in this context, we can drop it, right.

Comment (codingjoe) [reply tests/check_framework/test_model_checks.py:414]: I would keep the tests as atomic as possible. That aside, if you want to test the behavior for `abstract=False`, I would recommend dropping the PK and test that the check fails, hence the code inside the if-statement is actually being executed.

Comment (amureki) [reply tests/check_framework/test_model_checks.py]: @codingjoe very good point, Johannes :) I added a separate test to cover that check works in this case.

File: tests/check_framework/test_model_checks.py [hunk 2]
```diff
@@ -403,6 +403,29 @@ class Child(Parent):
...
         self.assertEqual(checks.run_checks(app_configs=self.apps.get_app_configs()), [])
 
+    def test_explicit_pk_with_abstract_parent(self):
+        class Abstract(models.Model):
+            class Meta:
+                abstract = True
+
+        class Child(Abstract):
+            id = models.AutoField(primary_key=True)
+
```
Comment (codingjoe) [inline tests/check_framework/test_model_checks.py:413]: This is unused now, besides that, it's looking good. However, I presume one of the follows has more ideas on how to improve the tests. They know the test suite better.

**Review by codingjoe — APPROVED** [review]
LGTM 👍

**Comment by claudep** [comment]
Isn't this a candidate to 3.2 backport?

**Comment by felixxm** [comment]
> Isn't this a candidate to 3.2 backport?

As far as I'm aware calling `.check()` on abstract models was never supported, but I don't see any harm in backport this. I will add release notes.

**Comment by felixxm** [comment]
@amureki Thanks for this patch :+1: I added release notes.

**Pull request merged**